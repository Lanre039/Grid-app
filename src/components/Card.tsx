import React from "react";
import { Items } from "../pages/types";

function Card({ shape, color }: Items): JSX.Element {
  const triangleBg = shape === "triangle" ? color : "transparent";
  return (
    <div className="main_bg">
      <div
        style={{
          borderBottomColor: triangleBg,
          borderBottomWidth: "81px",
          borderBottomStyle: "solid",
        }}
        className={`${shape} ${shape !== "triangle" ? color : ""}`}
      ></div>
    </div>
  );
}

export default Card;
