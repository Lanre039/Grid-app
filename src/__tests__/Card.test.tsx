import Adapter from "enzyme-adapter-react-16";
import { cleanup } from "@testing-library/react";
import { configure, shallow } from "enzyme";
import Card from "../components/Card";
configure({ adapter: new Adapter() });
afterEach(cleanup);

const renderCard = (shape: string, color: string) =>
  shallow(<Card shape={shape} color={color} />);

describe("<Card />", () => {
  it("render correctly", () => {
    const wrapper = renderCard("rectangle", "red");
    expect(wrapper).toMatchSnapshot();
  });

  it("should include color class when the shape is triangle", () => {
    const wrapper = renderCard("triangle", "red");
    expect(wrapper.find(".triangle")).toBeTruthy();
    expect(wrapper.find("triangle red")).toBeTruthy();
  });
});
