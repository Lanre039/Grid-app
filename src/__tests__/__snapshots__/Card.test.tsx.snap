// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> render correctly 1`] = `
<div
  className="main_bg"
>
  <div
    className="rectangle red"
    style={
      Object {
        "borderBottomColor": "transparent",
        "borderBottomStyle": "solid",
        "borderBottomWidth": "81px",
      }
    }
  />
</div>
`;
